python/dgl/data/minigc.py

```diff
@@ -61,12 +69,30 @@ def __getitem__(self, idx):
         """
         return self.graphs[idx], self.labels[idx]
 
+    def save(self):
+        """save the graph list and the labels"""
+        graph_path = os.path.join(self.raw_path, 'dgl_graph.bin')
+        # this check should be adeded into save_graphs
+        makedirs(self.raw_path)
```

Comment:
I suppose it is used in download. After calling download() self.raw_path should exist and contain raw data.

Reply:
Here are two things: 
1. Synthetic datasets (like this one) may not need to download from server.
2. `save_graphs`/`load_graphs` should have had the validation of the input path in python.

Reply:
> Here are two things:
> 
> 1. Synthetic datasets (like this one) may not need to download from server.
This can be a problem.

> 2. `save_graphs`/`load_graphs` should have had the validation of the input path in python.
Can we add a checkdir() other than makdir?

Reply:
1. We may need to walkthrough the current design to check if any variables require `download()` to initialize. 
2. The current `data.utils.makedir()` just works as check existance + `os.makedisr`.
3. `load_graphs` needs to check the existance of the file, but `save_graphs` needs to check the existance of the path of that file. E.g. given `/home/ubuntu/.dgl/minigc/graph.bin`, `save_graphs` need to check the validity of `/home/ubuntu/.dgl/minigc/`.